### PR TITLE
Adapt to new api

### PR DESCRIPTION
**Pull request by worm2fed**
Problem:
RandLabs updated API of Algorand node and indexer. For example, we know that `/health` and `/versions` endpoints apparently won't exist anymore.
Also they told that in future node will be used only for transactions.
    
Solution:
- Split Node and Indexer API types
- Add deprecation warning for some node commands and helpers
- Add few missing fields to indexer types and delete committed fields from node one
- Rename types for API
- Delete `/health` and `/versions` endpoints
- Add block endpoint to indexer along with helper function and CLA command.
- Block and Transaction data structures was added (it has same names with already presented at our sdk).
    
Note: 
for now there are no check that we operate at expected network, because we can not check it any more.
In `connectToNode` and `connectToIndexer` will return passed network without any check. We will make some changes to this when we receive an answer from RandLabs.

### CODE REVIEW

File: src/Network/Algorand/Api.hs [hunk 1]
```diff
@@ -7,12 +7,12 @@
 -- See <https://developer.algorand.org/docs/reference/rest-apis/algod/v2/>
```
Comment (gromakovsky) [inline src/Network/Algorand/Api.hs:7]: The comment above should be updated btw.

**Comment by gromakovsky** [comment]
> should I keep Maybe here util indexer will be fixed or we can just believe that it will be fixed soon?

I would keep `Maybe` until it's actually fixed. I'd like us to update `bridge-web` to use version from this PR after it's merged, and we don't know how soon `bTransactions` will be made non-optional.

File: src/Network/Algorand/Api/Indexer.hs [hunk 2]
```diff
@@ -120,3 +123,61 @@ data IdxAccountResponse = IdxAccountResponse
   , iarCurrentRound :: Round
   } deriving stock (Generic, Show)
 $(deriveJSON algorandTrainOptions 'IdxAccountResponse)
+
+-- | Contains fields common to all transactions and serves as an envelope
+-- to all transactions type.
+data Transaction = Transaction
```
Comment (gromakovsky) [inline src/Network/Algorand/Api/Indexer.hs:129]: Maybe let's call it `TransactionIdx` or `IdxTransaction` for example? To make it clearer that it's transaction for interaction with the indexer. Same for `Block`.

File: src/Network/Algorand/Api.hs [hunk 3]
```diff
@@ -78,10 +78,14 @@ data NodeApi route = NodeApi
...
+data IndexerApi route = IndexerApi
   { _accountIdx :: route
       :- "accounts"
       :> Capture "address" Address
       :> QueryParam "round" Round
       :> Get '[JSON] IdxAccountResponse
+  , _blockIdx :: route
+      :- "blocks"
+      :> Capture "round" Round
+      :> Get '[JSON] Block
```
Comment (gromakovsky) [inline src/Network/Algorand/Api.hs:90]: Does it not accept format? Msgpack should be more efficient to process, so it'd be nice to use it instead of JSON if possible.

File: src/Network/Algorand/Api/Indexer.hs [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  , adDeleted :: Bool
+  -- ^ Whether or not this account is currently closed.
+  , adPendingRewards :: Microalgos
+  -- ^ amount of MicroAlgos of pending rewards in this account.
+  , adRewardBase :: Maybe Microalgos
+  -- ^ used as part of the rewards computation. Only applicable to accounts
+  -- which are participating.
+  , adRewards :: Microalgos
+  -- ^ total rewards of MicroAlgos the account has received, including pending rewards.
+  , adRound :: Word64
```
[the remark this answers is not in the document]
Comment (worm2fed) [reply src/Network/Algorand/Api/Indexer.hs:112]: Thanks, didn't noticed that

File: src/Network/Algorand/Api/Indexer.hs (hunk 2 above)
Comment (worm2fed) [reply src/Network/Algorand/Api/Indexer.hs:129]: In case of `Transaction` I still a bit confused, because node will be used to send transactions and indexer to retrieve almost all information.
What about `Block` - I thin that in future we will delete old definition.
I can name it `TransactionResp` and `BlockResp`.

Maybe it would be better to keep all this definitions (including `AccountData` and so on) at `Data` module.. not sure yet. Any way it's question of another issue / PR

File: src/Network/Algorand/Api.hs [hunk 5]
```diff
@@ -25,32 +25,21 @@ import Data.Algorand.Address (Address)
...
+  { _node :: route
       :- "v2"
-      :> ToServantApi ApiV2
-  , _idx2 :: route
+      :> ToServantApi NodeApi
+  , _indexer :: route
       :- "v2"
-      :> ToServantApi ApiIdx2
+      :> ToServantApi IndexerApi
   } deriving (Generic)
```
[the remark this answers is not in the document]
Comment (worm2fed) [reply src/Network/Algorand/Api.hs:39]: hm..
well we can keep this module just to re export all staff for `Api` and keep separate `NodeApi` and `IndexerApi` (without `Api` data) somewhere inside `Api/`

File: src/Network/Algorand/Api.hs (hunk 3 above)
Comment (gromakovsky) [reply src/Network/Algorand/Api.hs:90]: Hm, can you check whether passing `format=msgpack` works? Maybe it's just missing in documentation. If it doesn't work, can you please ask in #stakerdao-serokell-randlabs whether they will support it (and say that we'd like to use it because it's more efficient)?

File: src/Network/Algorand/Api/Indexer.hs (hunk 2 above)
Comment (gromakovsky) [reply src/Network/Algorand/Api/Indexer.hs:129]: > In case of `Transaction` I still a bit confused, because node will be used to send transactions and indexer to retrieve almost all information.

I think it's ok if transaction that you send and transaction that you receive are different. I just don't like that we have two types with the same name. `TransactionResp` should be ok for now I think.

File: src/Network/Algorand/Api.hs (hunk 3 above)
Comment (worm2fed) [reply src/Network/Algorand/Api.hs:90]:
```
{
  "message": "Unknown parameter detected: format"
}
```

File: src/Network/Algorand/Api.hs (hunk 1 above)
Comment (worm2fed) [reply src/Network/Algorand/Api.hs]: done
sorry for squashed commits

File: src/Network/Algorand/Api.hs (hunk 5 above)
[the remark this answers is not in the document]
Comment (worm2fed) [reply src/Network/Algorand/Api.hs]: done

File: src/Network/Algorand/Api/Indexer.hs (hunk 4 above)
[the remark this answers is not in the document]
Comment (worm2fed) [reply src/Network/Algorand/Api/Indexer.hs]: done

File: src/Network/Algorand/Api/Node.hs [hunk 6]
```diff
@@ -15,52 +13,22 @@ module Network.Algorand.Api.Type
...
-
--- | algod version information.
-data Version = Version
-  { vBuild :: BuildVersion
-  , vGenesisHashB64 :: Text
-  , vGenesisId :: Network
-  , vVersions :: [Text]
-  } deriving (Generic, Show)
-$(deriveJSON algorandSnakeOptions 'Version)
+import Network.Algorand.Api.Indexer (TealKeyValueStore)
```
[the remark this answers is not in the document]
Comment (worm2fed) [reply src/Network/Algorand/Api/Node.hs]: done

File: src/Network/Algorand/Api/Indexer.hs (hunk 2 above)
Comment (worm2fed) [reply src/Network/Algorand/Api/Indexer.hs]: I've found a way to delete duplicate `Transaction` - there were a lot of problems with this duplication at bridge..
so, actually we do not need it, we need only correct json instances.

Can you help? How can I derive these instances for `TransactionType`?

[data TransactionType](https://github.com/serokell/haskell-algorand-sdk/blob/8758baa9a96be18329384802b1f346e2effe80a9/src/Data/Algorand/Transaction.hs#L76-L109)

File: src/Network/Algorand/Api.hs (hunk 3 above)
Comment (worm2fed) [reply src/Network/Algorand/Api.hs]: > Mauro Leggieri  ahhh, no, the algorand indexer and our only support json

File: src/Network/Algorand/Api/Indexer.hs (hunk 2 above)
Comment (worm2fed) [reply src/Network/Algorand/Api/Indexer.hs]: https://developer.algorand.org/docs/reference/rest-apis/indexer/#transaction

as I understand Aeson derives `tx-type` like this:
```
{
  ...
  tx-type: {
    ...
  },
  ...
}
```

but I need this:
```
{
  ...
  tx-type: pay,
  payment-transaction: {
    ...
  },
  ...
}
```

Note, that `pay` and `payment-transaction` is example and there are different transaction types.
Is it possible to derive json instances this way?

Comment (worm2fed) [reply src/Network/Algorand/Api/Indexer.hs]: I think I can split this sum type to one have only type labels and others to keep special transaction data
My only concern that we use current type to build transactions

Comment (worm2fed) [reply src/Network/Algorand/Api/Indexer.hs]: hm...
https://developer.algorand.org/docs/features/transactions/

according to this we actually just need to put string to specify transaction type

Comment (worm2fed) [reply src/Network/Algorand/Api/Indexer.hs]: I can leave definition as it is and manually add json instances like this:

```
instance ToJSON TransactionType where
  toJSON txType = toJSON (value :: Text)
    where
      value = case txType of
        PaymentTransaction{} -> "pay"
        ApplicationCallTransaction{} -> "appl"
        AssetTransferTransaction{} -> "axfer"
        KeyRegistrationTransaction -> "keyreg"
        AssetConfigTransaction -> "acfg"
        AssetFreezeTransaction -> "afrz"

instance FromJSON TransactionType where
  parseJSON o = do
    value :: Text <- parseJSON o
    pure $ case value of
      "pay" -> PaymentTransaction{}
      "appl" -> ApplicationCallTransaction{}
      "axfer" -> AssetTransferTransaction{}
      "keyreg" -> KeyRegistrationTransaction
      "acfg" -> AssetConfigTransaction
      "afrz" -> AssetFreezeTransaction
      x -> error . T.unpack $ "Unmapped transaction type field name: " <> x
 ```
 
 but this way some records not initialised

**Comment by worm2fed** [comment]
@gromakovsky I have one more question.
Earlier we exported some types without constructors - as I know this is architecture approach when we create kinda DSL and provide some function to retrieve or create data. 
I can add such helper functions to keep this approach. What do you think?

File: app/halgo/Halgo/CLA/Command/Contract.hs [hunk 7]
```diff
@@ -18,6 +18,8 @@ import qualified Data.Algorand.Address as A
 import qualified Network.Algorand.Api as Api
 
 import Network.Algorand.Definitions (Host)
+import Data.Algorand.Teal (TealCode (..), TealCompilationResult (..))
+
```
Comment (gromakovsky) [inline app/halgo/Halgo/CLA/Command/Contract.hs:22]: Looks like an extra newline.

File: app/halgo/Halgo/CLA/Command/Node.hs [hunk 8]
```diff
@@ -122,6 +114,7 @@ cmdPrintBlock rnd url = withNode url $ \(_, api) -> do
       mapM_ putJson txs
     _ -> putTextLn "No block found"
 
+{-# DEPRECATED cmdNodeFetchAccount "Use `cmdIndexerFetchAccount` instead" #-}
```
Comment (gromakovsky) [inline app/halgo/Halgo/CLA/Command/Node.hs:119]: I find it strange to have `DEPRECATED` pragma in **executable**. The only possible user of this function is this executable, it's not exported externally. Are we using this function in this executable? If no, we can remove it, there is no user and can't be any.

File: src/Network/Algorand/Client.hs [hunk 9]
```diff
@@ -9,30 +9,22 @@ module Network.Algorand.Client
...
-import Servant.Client (ClientEnv, mkClientEnv, parseBaseUrl, runClientM)
+import Servant.Client (mkClientEnv, parseBaseUrl, runClientM)
 import Servant.Client.Generic (AsClientT, genericClientHoist)
 
-import qualified Network.Algorand.Api as Api
-
-import Network.Algorand.Api (Api (..), ApiAny (..), ApiIdx2, ApiV2, Version (..))
-import Network.Algorand.Definitions (Host, Network, NetworkError (WrongNetwork))
-
-apiClient :: forall m' . MonadIO m' => ClientEnv -> Api (AsClientT m')
```
Comment (gromakovsky) [inline src/Network/Algorand/Client.hs:30]: Why did you remove `apiClient` helper? Now I see `genericClientHoist $ \x -> liftIO $ runClientM x env >>= either throwM pure` in 2 places.

File: src/Network/Algorand/Util.hs [hunk 10]
```diff
@@ -108,11 +132,31 @@ lookupAppLocalState Api.Account{..} appId = do
...
         "lookupAppLocalState: unknown teal value type "
-        <> show (Api.tvType tkeValue)
+        <> show (tvType tkeValue)
+
+-- | Helper to get account local state from indexer
+getAppLocalState
+  :: Api.AccountData
+  -> AppIndex
+  -> Maybe (Map ByteString (Either ByteString Word64))
+getAppLocalState Api.AccountData{..} appId = do
```
Comment (gromakovsky) [inline src/Network/Algorand/Util.hs:162]: Can you please extract common logic between `lookupAppLocalState` and `getAppLocalState` into a separate function?

File: src/Data/Algorand/MessagePack/Json.hs [hunk 11]
```diff
@@ -16,31 +16,33 @@ module Data.Algorand.MessagePack.Json
...
+import Data.ByteString (ByteString)
+import Data.ByteString.Base64 (decodeBase64, encodeBase64)
+import Data.Foldable (toList)
 import Data.Text (Text)
 import Data.Text.Encoding (encodeUtf8)
 
 import qualified Data.Algorand.Address as A
-import Data.Algorand.MessagePack (Canonical (Canonical), EitherError (..), MessagePackObject, MessageUnpackObject, fromAlgoObject, toAlgoObject)
 
+import Data.Algorand.MessagePack (Canonical (Canonical), EitherError (..), MessagePackObject, MessageUnpackObject, fromAlgoObject, toAlgoObject)
```
Comment (gromakovsky) [inline src/Data/Algorand/MessagePack/Json.hs:31]: Looks too big. Why is it not auto-formatted?

File: src/Data/Algorand/Transaction.hs [hunk 12]
```diff
@@ -106,7 +107,29 @@ data TransactionType
...
+  parseJSON o = do
+    value :: Text <- parseJSON o
+    pure $ case value of
+      "pay" -> PaymentTransaction{}
+      "appl" -> ApplicationCallTransaction{}
+      "axfer" -> AssetTransferTransaction{}
+      "keyreg" -> KeyRegistrationTransaction
+      "acfg" -> AssetConfigTransaction
+      "afrz" -> AssetFreezeTransaction
+      x -> error . T.unpack $ "Unmapped transaction type field name: " <> x
```
Comment (gromakovsky) [inline src/Data/Algorand/Transaction.hs:132]: Please use `fail` in `parseJSON`.

**Review by gromakovsky — CHANGES_REQUESTED** [review]

File: src/Network/Algorand/Client.hs (hunk 9 above)
Comment (worm2fed) [reply src/Network/Algorand/Client.hs:30]: because you asked to delete `API` type which combine node and indexer api's) 
I thought about type signature of `apiClient` and didn't found solution without `API` type. Any ideas?

File: src/Network/Algorand/Util.hs (hunk 10 above)
Comment (worm2fed) [reply src/Network/Algorand/Util.hs:162]: Yes, I can. But my logic was, that in future we are going to delete `lookupAppLocalState` (because actually this is same functions, but one for node and other one for indexer)

File: src/Data/Algorand/MessagePack/Json.hs (hunk 11 above)
Comment (worm2fed) [reply src/Data/Algorand/MessagePack/Json.hs:31]: hm.. sometimes hls feels bad (and formatter is built-in to hls). I'll check it

File: src/Data/Algorand/Transaction.hs (hunk 12 above)
Comment (worm2fed) [reply src/Data/Algorand/Transaction.hs:132]: ops.. I sure I've changed this copy paste to `fail`, maybe `cmd + z`. I'll do it

File: src/Network/Algorand/Api.hs (hunk 3 above)
Comment (gromakovsky) [reply src/Network/Algorand/Api.hs]: Okay, that's quite sad. I was told by @kirelagin that our JSON parsers are fragile and we should test them carefully, they may work incorrectly in some cases. I'll take a deeper look into that.

File: src/Network/Algorand/Util.hs (hunk 10 above)
Comment (gromakovsky) [reply src/Network/Algorand/Util.hs:162]: I wanted to write "yes, but we don't know how soon we'll delete `lookupAppLocalState`, this duplication may cause some troubles before we delete deprecated stuff", but then I realized that nobody except for us really uses this library, so we can delete deprecated stuff pretty soon. So ok, let's not care about this duplication.

File: src/Network/Algorand/Api.hs (hunk 3 above)
Comment (worm2fed) [reply src/Network/Algorand/Api.hs]: okay

File: src/Network/Algorand/Client.hs (hunk 9 above)
Comment (worm2fed) [reply src/Network/Algorand/Client.hs:30]: I've made it)

File: src/Data/Algorand/MessagePack/Json.hs (hunk 11 above)
Comment (gromakovsky) [reply src/Data/Algorand/MessagePack/Json.hs]: AFAIK formatting is done by `stylish-haskell`, so you can try to invoke it manually from terminal and see what it says.

**Comment by gromakovsky** [comment]
Okay, so apparently the only remaining thing is about JSON, it will require a bit more thinking from me :slightly_smiling_face:

**Comment by worm2fed** [comment]
@gromakovsky and answer this question too, please
https://github.com/serokell/haskell-algorand-sdk/pull/13#issuecomment-898337637

**Comment by gromakovsky** [comment]
> I can add such helper functions to keep this approach.

For which types? I'd say it's not DSL, but simply hiding of underlying libraries under the interface (perhaps you mean the same by DSL here).

**Comment by worm2fed** [comment]
> For which types? I'd say it's not DSL, but simply hiding of underlying libraries under the interface (perhaps you mean the same by DSL here).

Yes, I know this is not DLS (but kind of). For example `Signature` or `SecretKey` types - there were some (and they are now too) helpers functions (which uses pattern match over this types, but they were at same module and module exported only helper functions and type without constructors).

For example:
``` Haskell
module A
  ( Signature
  , function
  , ...
  ) where
type Signature = Signature Text

function (Signature s) = ...
```

File: src/Network/Algorand/Api/Indexer.hs (hunk 2 above)
Comment (gromakovsky) [reply src/Network/Algorand/Api/Indexer.hs]: > I've found a way to delete duplicate Transaction - there were a lot of problems with this duplication at bridge..
so, actually we do not need it, we need only correct json instances.

Can you elaborate on that please? I see that previously we had
```
instance ToJSON Transaction where
  toJSON = toCanonicalJson

instance FromJSON Transaction where
  parseJSON = parseCanonicalJson
```

same for `TransactionType`. What's the problem with these instances (I can imagine they may be incorrect, but I'd like to know where the problem is and whether we can fix `toCanonicalJson` and `parseCanonicalJson` to make these things work)?

Anyway, I think we should parse `TransactionType` the same way as we do it for MsgPack. Specifically, `TransactionType` should be an object with mandatory `type` key  and other keys dependent on the `type`. Or we may have no instances for `TransactionType` and implement its encoding/decoding as part of `Transaction` encoding/decoding.
It seems impossible to construct suitable `SumEncoding` for this use case, so I would just write some boilerplate encoding/decoding code.

**Comment by gromakovsky** [comment]
Yes, I saw it, I am asking not about the types for which it's already the case (no constructor is exported), but about the types for which you want to do the same. If we are talking about types added in this PR, please tell me which ones to consider. If we are talking about old types, then I think it's definitely out of scope of this PR.

**Comment by worm2fed** [comment]
> If we are talking about types added in this PR, please tell me which ones to consider. If we are talking about old types, then I think it's definitely out of scope of this PR.

Yes, my question was about old types. 
Okay, we can discuss it later)

File: src/Network/Algorand/Api/Indexer.hs (hunk 2 above)
Comment (worm2fed) [reply src/Network/Algorand/Api/Indexer.hs]: For `Transaction` - in `msgpack` we use `type`, but indexer returning `tx-type` – `Could not decode response: Error in $.transactions[0]: Missing required key: type`
I think it's better to not change `msgpack` definition, because it's still relevant and such changes can break something.

For `TransactionType` I've got – `Could not decode response: Error in $.transactions[0]['tx-type']: invalid encoding for Assoc`

Comment (worm2fed) [reply src/Network/Algorand/Api/Indexer.hs]: In case I change `msgpack` to expect `tx-type` and keep old Json instances with canonical helper functions - it's working, but there are still missing objects, depending on `tx-type`

```json
{
    "transactions-root": "LN9CHN+zm1OHdoF2SPpftJW9YI2J/F0Zv/FJsNdAheA=",
    "transactions": [
        {
            "fee": 10000,
            "tx-type": "pay"
        }
    ],
    "seed": "sBoIA83cUpa8jNOdqmGClw85y0bYcFhfmKzrnxmhgdc=",
    "round": 1,
    "genesis-id": "testnet-v1.0",
    "rewards": {
        "rewards-level": 0,
        "rewards-pool": "7777777777777777777777777777777777777777777777777774MSJUVU",
        "rewards-residue": 250000000,
        "rewards-rate": 250000000,
        "fee-sink": "A7NMWS3NT3IUDMLVO26ULGXGIIOUQ3ND2TXSER6EBGRZNOBOUIQXHIBGDE",
        "rewards-calculation-round": 500000
    },
    "genesis-hash": "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=",
    "upgrade-state": {
        "next-protocol": "https://github.com/algorand/spec/tree/22726c9dcd12d9cddce4a8bd7e8ccaa707f74101",
        "next-protocol-switch-on": 20001,
        "current-protocol": "https://github.com/algorand/spec/tree/a26ed78ed8f834e2b9ccb6eb7d3ee9f629a6e622",
        "next-protocol-approvals": 1,
        "next-protocol-vote-before": 10001
    },
    "previous-block-hash": "G8RmwKHb8iQahvMCvwJVZuI97pdN/dvSIkr7ZCI0AlM=",
    "upgrade-vote": {
        "upgrade-approve": true,
        "upgrade-delay": 0,
        "upgrade-propose": "https://github.com/algorand/spec/tree/22726c9dcd12d9cddce4a8bd7e8ccaa707f74101"
    },
    "txn-counter": 0,
    "timestamp": 1560210480
}
```

Comment (worm2fed) [reply src/Network/Algorand/Api/Indexer.hs]: Interesting thing, that with deriving instance for `Transaction` we have more fields:

```json
{
    ...
    "transactions": [
        {
            "sender": "GD64YIY3TWGDMCNPP553DZPPR6LDUSFQOIJVFDPPXWEG3FVOJCCDBBHU5A",
            "fee": 10000,
            "last-valid": 1000,
            "tx-type": "pay",
            "first-valid": 0
        }
    ],
    ...
```
so, I guess there are not only `type` differ in `msgpack` format

Comment (gromakovsky) [reply src/Network/Algorand/Api/Indexer.hs]: > For Transaction - in msgpack we use `type`, but indexer returning `tx-type`

Hm, ok, what does node return if you request JSON from it? `type` or `tx-type`?

>  I think it's better to not change msgpack definition, because it's still relevant and such changes can break something.

Yes, let's not change msgpack. If JSON and msgpack are different, let's just define instances separately (without `toCanonicalJson`).

Comment (gromakovsky) [reply src/Network/Algorand/Api/Indexer.hs]: > In case I change msgpack to expect tx-type and keep old Json instances with canonical helper functions - it's working,

But how does it work if you say that in msgpack they have `type` instead of `tx-type`?

Comment (gromakovsky) [reply src/Network/Algorand/Api/Indexer.hs]: > Interesting thing, that with deriving instance for Transaction we have more fields:

Derived by TH or `toCanonicalJson`?

Comment (worm2fed) [reply src/Network/Algorand/Api/Indexer.hs]: > Hm, ok, what does node return if you request JSON from it? type or tx-type?
type (https://testnet.algoexplorerapi.io/v2/blocks/1?format=json)

> But how does it work if you say that in msgpack they have type instead of tx-type?
yes, yes. I've temporary changed `type` to `tx-type` to see what will be returned)

> Interesting thing, that with deriving instance for Transaction we have more fields
Derived by TH provide more fields than `toCanonicalJson`

**Comment by gromakovsky** [comment]
@worm2fed I've made some changes here and will push them shortly to this branch. Your version is saved to the `worm2fed/sdao-479-adapt-to-new-api-bk` branch just in case.

Overview of the changes:
1. Reworked commit history extracting some small changes into standalone commits.
2. Didn't add `AccountData` to indexer types, I think we can use existing `Account` there. I didn't find any essential differences. Also getting `Account` from node will be removed at some point and I guess we'll just move `Account` to the indexer (it won't be needed in the node).
3. Consequently, we don't need `getAppLocalState` and `getAssetBalance` anymore.
4. Fixed some things in README.
5. Added tests for parsing indexer responses.
6. Removed some fields that you added (e. g. in `BlockResp`) that we don't use anyway, because supporting them was not trivial.

**Review by gromakovsky — APPROVED** [review]
